docs/introduction.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Introduction
+id: introduction
+---
+
+# What is `granary`?
+
+Granary is a job runner for cloud-based geospatial machine learning.
+Its goal is to simplify running models and to track and serve
+the results of predictions. It puts a REST API between you and AWS Batch
+to simplify interactions that otherwise involve repeatedly checking AWS SDK
+documentation.
+
+# How to read this introduction
+
+This introduction includes `scala` language code blocks and their output. The
+output it automatically generated by the [`mdoc`](https://scalameta.org/mdoc/)
+tool. The steps described in this introduction include sending JSON to HTTP
+endpoints. While the JSON in the examples is generated automatically from
+`scala` values, clients can produce JSON using whatever languages and tools
+they desire.
+
+# Background info
+
+To begin with, here are some IDs and imports that we'll be using throughout the examples.
+They are present for consistency and compilation throughout the rest of this
+document. You can ignore them.
+
+```scala mdoc
+import com.rasterfoundry.granary.datamodel._
+import io.circe.syntax._
+import java.time.Instant
+import java.util.UUID
+
+val modelId = UUID.fromString("@MODEL_ID@")
+val predictionId = UUID.fromString("@PREDICTION_ID@")
+val webhookId = UUID.fromString("@WEBHOOK_ID@")
+val invocationTime = Instant.ofEpochMilli("@INVOCATION_TIME@".toLong)
+```
+
+The `modelId` is a faked ID that you can imagine represents a stored `Model`
+in the database. The same is true for the `predictionId`, but for a `Prediction`.
+The `webhookId` represents a single-use webhook that can be called to indicate
+the success or failure of a prediction.
+The `invocationTime` is a pretend time that this prediction was
+kicked off.
+
+# What's a `Model`?
+
+A model is a bundle of a human-readable name, some AWS Batch configuration,
+and an argument validator. `Model`s in Granary correspond to containers that
+have been configured to run via AWS Batch job definitions. The major difference
+between using Granary and hand rolling `SubmitJob` requests is the `Validator`.
+
+To create a model, `POST` JSON like this to `/api/models`:
+
+```scala mdoc
+Model.Create(
+  "A descriptive model name",
+  Validator(().asJson), // a Validator -- more on this below
+  "perfectAccuracyModel:1", // an AWS Batch job definition
+  "veryExpensiveOnDemandQueue" // an AWS Batch job queue
+).asJson.spaces2
```

Comment:
Does using something like [`passthrough`](https://scalameta.org/mdoc/docs/modifiers.html#passthrough) make sense here? That way we could print out the rendered json (and it'd be up to date forever!) which is the most important part for folks interacting with the API.

Similar comment below for the rest of these too

Reply:
I had no idea about `passthrough` or `invisible`. Those look great! I'll add them where appropriate.

Reply:
No more Scala is visible in the rendered docs. I had to do some tricks to get the formatting to come out nicely -- for example, the highlighter really hated rendering a multi-line empty object, which is how the empty json schema was printing -- but you can check out how it looks now on the live page